cvat-canvas/README.md

```diff
@@ -0,0 +1,93 @@
+## Module
+
+- Written on typescript
+- Contains the class ```Canvas```
+
+## Creation
+Canvas is created by using constructor:
+
+```js
+    const canvas = new Canvas();
+```
+
+- Canvas has transparent background
+
+Canvas itself handles:
+- Shape context menu (PKM)
+- Image moving (mousedrag)
+- Image resizing (mousewheel)
+- Image fit (dblclick)
+- Remove point (PKM)
+- Polyshape editing (Shift + LKM)
+
+## API
+### Methods
+
+All methods are sync.
+
+```js
+    html() => canvas HTML element
+    setup(const FrameData: frameData, [{
+        state: ObjectState,
+        appearance: {   // all these fields are optional
+            borderColor: 'color',
+            fillColor: 'color',
+            fillOpacity: 50,
+            selectedFillOpacity: 50,
+        }
+    }]) => undefined
+
+    activate(const number: clientID, const number: attributeID = null) => undefined // select if can't activate
+    rotate(const number: degrees = 90) => undefined
+    focus(const number: id, const number: padding) => undefined
+    fit() => undefined
+    grid(stepX, stepY, color, opacity) => undefined
+    adjust({
```

Comment:
Need to have access to your image as well. In the future we can "external" ways to modify Image itself. Also probably instead of "explicit" adjust method you can have a css class for your image and UI will change the image using the class.

I see here common pattern: appearance of elements should be changed using CSS instead of explicit methods. I believe the approach is more flexible. Logic/behaviours of cvat-canvas should be changed using methods.

Reply:
Done